src/vs/workbench/contrib/chat/browser/actions/reusablePromptActions/dialogs/askToSelectPrompt/utils/attachPrompt.ts

```diff
@@ -78,36 +100,67 @@ const isAttachedAsCurrentPrompt = (
 };
 
 /**
- * Attaches provided prompts to a chat input.
+ * Attaches provided instructions to a chat input.
  */
-export const attachPrompt = async (
-	file: URI,
-	options: IAttachPromptOptions,
+export const attachInstructionsFiles = async (
+	files: URI[],
+	options: IAttachOptions,
 ): Promise<IAttachResult> => {
-	const { skipIfImplicitlyAttached } = options;
 
 	const widget = await getChatWidgetObject(options);
 
-	if (skipIfImplicitlyAttached && isAttachedAsCurrentPrompt(file, widget)) {
-		return { widget, wasAlreadyAttached: true };
-	}
+	const wasAlreadyAttached: URI[] = [];
 
-	const wasAlreadyAttached = widget
-		.attachmentModel
-		.promptInstructions
-		.add(file);
+	for (const file of files) {
+		if (widget.attachmentModel.promptInstructions.add(file)) {
+			wasAlreadyAttached.push(file);
+			continue;
+		}
+	}
 
 	return { widget, wasAlreadyAttached };
 };
 
 /**
- * Gets a chat widget based on the provided {@link IChatAttachPromptActionOptions.widget widget}
+ * Runs the prompt file
+ */
+export const runPromptFile = async (
```

Comment:
This function is out of place in this file, can be moved to `runPrompt.ts` or similar. (or at least to `chatRunPromptAction.ts`?)

Reply:
Its now in its own file